transcode/transcode.go

```diff
@@ -94,6 +94,34 @@ func RunTranscodeProcess(transcodeRequest TranscodeSegmentRequest, streamName st
 		return outputs, segmentsCount, fmt.Errorf("error generating source segment URLs: %s", err)
 	}
 
+	// The last segment in an HLS manifest may contain an audio-only track - this is common
+	// during a livestream recording where the video stream can end sooner with a trailing audio stream
+	// which results in a segment at the end that just contains audio. This segment should *not* be
+	// submitted to the T.
+	lastSegment := sourceSegmentURLs[len(sourceSegmentURLs)-1]
+	lastSegmentURL, err := clients.SignURL(lastSegment.URL)
+	if err != nil {
+		return outputs, segmentsCount, fmt.Errorf("failed to create signed url for last segment %s: %w", lastSegment.URL, err)
+	}
+	p := video.Probe{}
+	// ProbeFile will return err for various reasons so we use the subsequent GetTrack method to check for video tracks
+	lastSegmentProbe, _ := p.ProbeFile(transcodeRequest.RequestID, lastSegmentURL)
+	// GetTrack will return an err if TrackTypeVideo was not found
+	_, err = lastSegmentProbe.GetTrack(video.TrackTypeVideo)
+	if err != nil {
+		var lastSegmentIdx int
+		for i, entry := range sourceManifest.Segments {
+			if entry == nil {
+				lastSegmentIdx = i - 1
+				break
+			}
+		}
```

Comment:
Thanks for leaving this comment @thomshutt here - a simple change to drop an element from this `Segments` slice was giving me a headache and had me questioning my existence and why this slice length was pre-allocated 1024 bytes: 
https://github.com/livepeer/catalyst-api/blob/main/clients/manifest.go#L69

Reply:
There is a helper func btw: `GetAllSegments()` so you don't have to remember.